kunta-api-server/src/main/java/fi/otavanopisto/kuntaapi/server/integrations/mikkelinyt/MikkeliNytEventProvider.java

```diff
@@ -412,4 +485,62 @@ private LocalDateTime parseLocalDateTime(String text) {
     return LocalDateTime.parse(text, formatter);
   }
 
+  private class EventComparator implements Comparator<fi.otavanopisto.mikkelinyt.model.Event> {
+    
+    private EventOrder order;
+    private EventOrderDirection direction;
+    
+    public EventComparator(EventOrder order, EventOrderDirection direction) {
+      this.order = order;
+      this.direction = direction;
+    }
+    
+    @Override
+    public int compare(fi.otavanopisto.mikkelinyt.model.Event event1, fi.otavanopisto.mikkelinyt.model.Event event2) {
+      int result;
+      
+      switch (order) {
+        case END_DATE:
+          result = compareEndDates(event1, event2);
+        break;
+        case START_DATE:
+          result = compareStartDates(event1, event2);
+        break;
+        default:
+          result = 0;
+        break;
+      }
+      
+      if (direction == EventOrderDirection.ASCENDING) {
+        return -result;
+      } 
+      
+      return result;
+    }
+
+    private int compareStartDates(fi.otavanopisto.mikkelinyt.model.Event event1,
+        fi.otavanopisto.mikkelinyt.model.Event event2) {
+      return compareDates(parseLocalDateTime(event1.getStart()), parseLocalDateTime(event2.getStart()));
+    }
+
+    private int compareEndDates(fi.otavanopisto.mikkelinyt.model.Event event1,
+        fi.otavanopisto.mikkelinyt.model.Event event2) {
+      return compareDates(parseLocalDateTime(event1.getEnd()), parseLocalDateTime(event2.getEnd()));
+    }
+    
+    private int compareDates(LocalDateTime dateTime1, LocalDateTime dateTime2) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "compareDates" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)